diff --git a/drivers/misc/mediatek/mdp/mdp_ioctl_ex.c b/drivers/misc/mediatek/mdp/mdp_ioctl_ex.c
index 4a3880cfce47..1a3404321810 100644
--- a/drivers/misc/mediatek/mdp/mdp_ioctl_ex.c
+++ b/drivers/misc/mediatek/mdp/mdp_ioctl_ex.c
@@ -29,7 +29,7 @@
 #include "mdp_cmdq_record.h"
 #include "mdp_cmdq_device.h"
 #else
-#include "cmdq_def.h"
+#include "../cmdq/v3/cmdq_def.h"
 #include "cmdq_mdp_common.h"
 #include "cmdq_driver.h"
 #include "cmdq_record.h"
