diff --git a/arch/arm64/boot/dts/mediatek/mt6768.dts b/arch/arm64/boot/dts/mediatek/mt6768.dts
index 394b0cf69b43..28a5334b31f1 100644
--- a/arch/arm64/boot/dts/mediatek/mt6768.dts
+++ b/arch/arm64/boot/dts/mediatek/mt6768.dts
@@ -5105,3 +5105,5 @@ defined(CONFIG_MTK_CAM_SECURITY_SUPPORT)
 #ifdef CONFIG_MTK_ENABLE_GENIEZONE
 #include "mediatek/trusty.dtsi"
 #endif
+
+#include "mediatek/lancelot.dtsi"
