From 5662d4484201fcf0626c2a515d00c5bef7109c19 Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Thu, 20 Mar 2025 00:46:54 -0400
Subject: [PATCH] move autogen to use env sh instead of hardcoded bash path

---
 autogen.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autogen.sh b/autogen.sh
index b74f544..d4cc868 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env sh
 
 #cleanup cache for correct versioning when run multiple times
 rm -rf autom4te.cache
-- 
2.48.1

