From 8acdd5d402a9963ad1f813050a5d815e4b27ccb6 Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Tue, 25 Mar 2025 19:36:31 -0400
Subject: [PATCH] revert dbe8d48

---
 pyimg4/parser.py | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/pyimg4/parser.py b/pyimg4/parser.py
index cde4b93..0053607 100644
--- a/pyimg4/parser.py
+++ b/pyimg4/parser.py
@@ -365,22 +365,8 @@ class IM4M(_PyIMG4):
         for _ in range(4):
             self._decoder.leave()
 
-        if self._decoder.eof():
-            # IM4M has no signature/certificates
-            # TODO: Make this cleaner
-            self._signature = None
-            self._certificates = None
-            return
-
-        if self._decoder.peek().nr != asn1.Numbers.OctetString:
-            raise UnexpectedTagError(self._decoder.peek(), asn1.Numbers.OctetString)
-
         self._signature = self._decoder.read()[1]
 
-        if self._decoder.peek().nr != asn1.Numbers.Sequence:
-            raise UnexpectedTagError(self._decoder.peek(), asn1.Numbers.Sequence)
-
-        # TODO: Parse certificates
         self._certificates = self._decoder.read()[1]
 
         if not self._decoder.eof():
-- 
2.49.0

