From 3b8ae546bc9f7e911f9dce04f219c621d6220baf Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Fri, 5 Jul 2024 10:38:50 -0700
Subject: [PATCH 4/4] abuild: add warning in -systemd split func about
 non-usr-merged birs

---
 abuild.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/abuild.in b/abuild.in
index 937c3e9..e0a4e63 100644
--- a/abuild.in
+++ b/abuild.in
@@ -2211,6 +2211,10 @@ default_systemd() {
 		return 1
 	fi
 
+	if [ -d bin ] || [ -d lib ] || [ -d usr/sbin ]; then
+		warning "Files should be installed to /usr/bin or /usr/lib, not /bin, /lib, or /usr/sbin"
+	fi
+
 	amove usr/lib/systemd
 
 	# Find user and system unit files, for use later when generating install scripts
-- 
2.48.1

