diff -ru fontconfig-2.15.0.orig/conf.d/60-latin.conf fontconfig-2.15.0/conf.d/60-latin.conf
--- fontconfig-2.15.0.orig/conf.d/60-latin.conf	2021-12-03 19:48:08.000000000 +0900
+++ fontconfig-2.15.0/conf.d/60-latin.conf	2024-09-27 05:19:21.398135978 +0900
@@ -6,6 +6,7 @@
 		<family>serif</family>
 		<prefer>
 			<family>Noto Serif</family>
+			<family>Droid Serif</family>
 			<family>DejaVu Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
@@ -19,6 +20,7 @@
 		<family>sans-serif</family>
 		<prefer>
 			<family>Noto Sans</family>
+			<family>Droid Sans</family>
 			<family>DejaVu Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
@@ -36,6 +38,7 @@
 		<family>monospace</family>
 		<prefer>
 			<family>Noto Sans Mono</family>
+			<family>Droid Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
